app/api/activitylog/activitylogMiddleware.js

```diff
@@ -32,26 +34,40 @@ function mustBeLogged(baseurl, method, body) {
   return isLoggedRequest && validBody;
 }
 
-export default (req, _res, next) => {
+const createEntry = req => {
   const { url, method, params, query, body, user = {} } = req;
-  const baseurl = url.split('?').shift();
-  if (mustBeLogged(baseurl, method, body)) {
-    const expireAt = date.addYearsToCurrentDate(1);
-    const bodyLog = { ...body };
-    if (bodyLog.password) bodyLog.password = '*****';
-    const entry = {
-      url: baseurl,
-      method,
-      params: JSON.stringify(params),
-      query: JSON.stringify(query),
-      body: JSON.stringify(bodyLog),
-      user: user._id,
-      username: user.username,
-      time: Date.now(),
-      expireAt,
-    };
-    activitylog.save(entry);
-    appendFile(activityLogPath(), `${JSON.stringify(entry)}\r\n`);
+  const expireAt = date.addYearsToCurrentDate(1);
+  const bodyLog = { ...body };
+  if (bodyLog.password) bodyLog.password = '*****';
+  return {
+    url,
+    method,
+    params: JSON.stringify(params),
+    query: JSON.stringify(query),
+    body: JSON.stringify(bodyLog),
+    user: user._id,
+    username: user.username,
+    time: Date.now(),
+    expireAt,
+  };
+};
+
+export default (req, _res, next) => {
+  try {
+    const { url, method, body = {} } = req;
+    const baseurl = url.split('?').shift();
+    if (mustBeLogged(baseurl, method, body)) {
+      const entry = createEntry({ ...req, url: baseurl });
+      activitylog.save(entry);
+      appendFile(
+        activityLogPath(`${tenants.current().name}$_activity.log`),
```

Comment:
@Zasa-san i think the ```$``` here is a typo ? the generated file is ```tenant$_activity.log``` ?

Reply:
Yes, its a typo. Fixed.